compiler/src/source_map.rs

```diff
@@ -19,6 +19,27 @@ impl Files {
         Self::default()
     }
 
+    pub fn from_dirs(paths: &Vec<PathBuf>, filter: SourceFilter) -> Result<Self, Error> {
```

Comment:
could accept any slice, not just a Vec
```suggestion
    pub fn from_dirs(paths: &[PathBuf], filter: SourceFilter) -> Result<Self, Error> {
```

Reply:
ahh, gotcha!